grpc/rust/BUILD

```diff
@@ -0,0 +1,64 @@
+load("@rules_rust//proto:proto.bzl", "rust_grpc_library")
+load("@rules_rust//proto:toolchain.bzl", "rust_proto_toolchain")
+load("@rules_rust//rust:defs.bzl", "rust_binary")
+
+# TODO: once we clarify how we should compile the gRPC + Protobuf library, delete this target
+rust_binary(
+    name = "proto_compile",
+    srcs = ["main.rs"],
+    data = [
+        "//cluster:service-proto",
+        "//cluster:server-proto",
+        "//cluster:user-proto",
+        "//cluster:database-proto",
+        "//common:answer-proto",
+        "//common:concept-proto",
+        "//common:logic-proto",
+        "//common:options-proto",
+        "//common:query-proto",
+        "//common:session-proto",
+        "//common:transaction-proto",
+        "//core:database-proto",
+        "//core:service-proto",
+    ],
+    deps = [
+        "@vaticle_dependencies//library/crates:protoc_rust_grpc",
+    ],
+)
+
+rust_proto_toolchain(
+    name = "proto_toolchain_impl",
+    # Path to the protobuf compiler.
+    protoc = "@com_google_protobuf//:protoc",
+    # Protobuf compiler plugin to generate rust gRPC stubs.
+    grpc_plugin = "@vaticle_dependencies//library/crates:cargo_bin_protoc_gen_rust_grpc",
+    # Protobuf compiler plugin to generate rust protobuf stubs.
+    proto_plugin = "@vaticle_dependencies//library/crates:cargo_bin_protoc_gen_rust",
+)
+
+toolchain(
+    name = "proto_toolchain",
+    toolchain = ":proto_toolchain_impl",
+    toolchain_type = "@rules_rust//proto:toolchain_type",
+)
+
+rust_grpc_library(
```

Comment:
This is our preferred way of creating a Rust gRPC library, and is symmetric with our Java, Python and NodeJS implementations. We delegate the work fully to `rules_rust`.

The only limitation is that it uses the `grpc` crate. Ideally, we would use the `tonic` crate, but I couldn't find a prebuilt Bazel rule that uses it.